RandBLAS/random_gen.hh

```diff
@@ -0,0 +1,20 @@
+#ifndef random_gen_hh
+#define random_gen_hh
+
+// this is for sincosf
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <math.h>
+#include <Random123/array.h>
+#include <Random123/philox.h>
+#include <Random123/uniform.hpp>
+#include <Random123/philox.h>
+#include <Random123/threefry.h>
+#include <Random123/boxmuller.hpp>
+
+using Philox = r123::Philox4x32;
+using Threefry = r123::Threefry4x32;
```

Comment:
Earlier I used typedefs to create aliases for these types. How does this ``using`` assignment compare to a typedef? From a source-code perspective, I'm guessing there's no real difference. Does it result in less-messy compiler errors?

Reply:
A typedef and an alias are different features. Aliases can be templated while typedefs cannot.  Otherwise they are equivalent. my opinion is that as part of our code style we might use aliases every where for consistency sake. a minor issue though.

Reply:
Gotcha. I just looked into aliases vs typedefs, and I agree with your conclusion.